pkg/k8sutil/create.go

```diff
@@ -227,3 +230,88 @@ func createNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) erro
 func ListNamespaces(nsclient corev1typed.NamespaceInterface) (*v1.NamespaceList, error) {
 	return nsclient.List(context.TODO(), metav1.ListOptions{})
 }
+
+// ObjectMetadata uniquely identifies any object in the list of YAML manifests.
+type ObjectMetadata struct {
+	Version string
+	Kind    string
+	Name    string
+}
+
+// YAMLToUnstructured accepts a Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata from the Kubernetes manifest.
+func YAMLToUnstructured(yamlObj []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode(yamlObj, nil, u); err != nil {
+		return nil, fmt.Errorf("converting config to unstructured.Unstructured: %w", err)
+	}
+
+	return u, nil
+}
+
+// removeYAMLComments converts YAML to JSON and back again, this removes the comments in the YAML
+// and any extra whitespaces spaces.
+func removeYAMLComments(yamlObj []byte) ([]byte, error) {
+	jsonObj, err := sigyaml.YAMLToJSON(yamlObj)
+	if err != nil {
+		return nil, fmt.Errorf("converting YAML to JSON: %w", err)
+	}
+
+	yamlObj, err = sigyaml.JSONToYAML(jsonObj)
+	if err != nil {
+		return nil, fmt.Errorf("converting JSON to YAML: %w", err)
+	}
+
+	return yamlObj, nil
+}
+
+// SplitYAMLDocuments converts a YAML string with multiple YAML docs separated by `---` into unique
+// objects and returns those objects as a map.
+func SplitYAMLDocuments(yamlObj string) (map[ObjectMetadata]string, error) {
```

Comment:
This function does more than splitting, as it also converts it to `ObjectMetadata`. Name of the function does not reflect that. I'd expect `SplitYAMLDocuments(yamlObj string)` to return `[]string`. which can be then piped to `YAMLToUnstructured`.

Reply:
Also, after splitting we could then change `parseManifests` to convert to `manifest`, so we don't have 2 implementation of reading YAML files and removing comments from them, which was my original goal.

Reply:
Changed the func to not return the map of objects rather just split the YAML file.